magneticod/magneticod/dht.py

```diff
@@ -30,107 +31,95 @@
 
 
 class SybilNode:
-    def __init__(self, address: typing.Tuple[str, int]):
+    def __init__(self, address: typing.Tuple[str, int], complete_info_hashes, max_metadata_size):
         self.__true_id = self.__random_bytes(20)
 
-        self.__socket = socket.socket(type=socket.SOCK_DGRAM)
-        self.__socket.bind(address)
-        self.__socket.setblocking(False)
+        self.__address = address
 
-        self.__incoming_buffer = array.array("B", (0 for _ in range(65536)))
-        self.__outgoing_queue = collections.deque()
-
-        self.__routing_table = {}  # type: typing.Dict[NodeID, NodeAddress]
+        self._routing_table = {}  # type: typing.Dict[NodeID, NodeAddress]
 
         self.__token_secret = self.__random_bytes(4)
         # Maximum number of neighbours (this is a THRESHOLD where, once reached, the search for new neighbours will
         # stop; but until then, the total number of neighbours might exceed the threshold).
         self.__n_max_neighbours = 2000
+        self.__tasks = {}  # type: typing.Dict[dht.InfoHash, asyncio.Future]
+        self._complete_info_hashes = complete_info_hashes
+        self.__max_metadata_size = max_metadata_size
+        self._metadata_q = asyncio.Queue()
+        self._is_paused = False
+        self._tick_task = None
 
         logging.info("SybilNode %s on %s initialized!", self.__true_id.hex().upper(), address)
 
-    @staticmethod
-    def when_peer_found(info_hash: InfoHash, peer_addr: PeerAddress) -> None:
-        raise NotImplementedError()
+    async def launch(self, loop):
+        self._loop = loop
```

Comment:
Yes, you can do that. I think in general, it's better practice to explicitly pass along the current loop, but we could certainly get away with not doing that here.